test/jdk/java/lang/reflect/code/writer/TestOpMethod.java

```diff
@@ -0,0 +1,163 @@
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.lang.classfile.*;
+import java.lang.classfile.components.ClassPrinter;
+import java.lang.classfile.constantpool.PoolEntry;
+import java.lang.classfile.constantpool.StringEntry;
+import java.lang.classfile.instruction.*;
+import java.lang.constant.ConstantDescs;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.code.Op;
+import java.lang.reflect.code.bytecode.BytecodeGenerator;
+import java.lang.reflect.code.interpreter.Interpreter;
+import java.lang.reflect.code.op.ExtendedOp;
+import java.lang.reflect.code.parser.OpParser;
+import java.lang.reflect.code.type.CoreTypeFactory;
+import java.lang.reflect.code.type.MethodRef;
+import java.lang.reflect.code.writer.OpBuilder;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.reflect.code.op.CoreOp.FuncOp;
+
+/*
+ * @test
+ * @enablePreview
+ * @run testng TestOpMethod
+ *
+ */
+public class TestOpMethod {
+
+//    We discussed replacing the mechanism to store models in class files.
+//    Currently, we serialize to the textual form.
+//    We want to explore replacing that with methods in class files that build and return models.
+//    See the test TestCodeBuilder and familiarize yourself with that.
+//    It's likely that transformation code is buggy because it has not been tested on a wide range of source.
+//    It would also be interesting to get some size comparison between the two approaches.
+//    To properly do this we will need to avoid the copying of code from java.base into jdk.compiler,
+//    otherwise we need to copy more code and there are certain restrictions on what features the code can use.
+//    Maurizio is looking into that.
+//    There may be an interim solution before moving the code to an incubating model. We should discuss more in our meetings.
+//    One way to explore sooner for experimentation purposes to write a class file transformer using the Classfile API
+//    and rewrite the class file replacing code models in textual form with the corresponding methods.
+//    That's more challenging when reflecting over lambda bodies, but should more feasible when reflecting over method bodies.
+
+    static ClassModel CLASS_MODEL;
+    @BeforeClass
+    static void setup() throws IOException {
+        CLASS_MODEL = ClassFile.of().parse(IR.class.getResourceAsStream("IR.class").readAllBytes());
+    }
+
+    @DataProvider
+    byte[][] classes() {
+        try {
+            return new byte[][] {
+                    IR.class.getResourceAsStream("IR.class").readAllBytes()
+            };
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test(dataProvider = "classes")
+    void test(byte[] cd) throws Throwable {
```

Comment:
That's a good start, but i suspect what we need here is also a main method, so we can run this over many classes via some script. Thereby we get more (albeit manual) test exposure and can get statistics class file sizes before and after i.e. we can run the script over all the compiled test class files in lang and jdk.